src/components/MultiSelect/MultiSelect.stories.mdx

```diff
@@ -0,0 +1,322 @@
+          name: "Child 2",
+        },
+      ],
+    },
+  ]}
+/>
+```
+
+## Controlling state using selectedItems and onChange prop
+
+While the component does not store the selectedItems state internally, it expects
+a certain state shape to be used for calculating the selected item count. This
+state shape is a "normalized" pattern, and is basically an object of objects,
+keyed by the multiselect id.
+
+```json
+{
+  "subjects": {
+    "items": ["art", "architecture"]
+  },
+  "locations": {
+    "items": ["sasb"]
+  }
+}
+```
+
+This is helpful for when several MultiSelect components are rendered inside a FilterBar component,
+you can use the MultiSelect id to get items for a specific slice of the state.
+
+```jsx
+const subjectsSelectedItemssOnly = selectedItems["subjects"].items;
+const locationsSelectedItemssOnly = selectedItems["locations"].items;
+```
+
+It's also helpful if a consuming app is pushing state into the browser's url.
+This example uses NextJS's router push method to add query parameters to url.
+
+```jsx
+const multiSelectId = "subjects";
+// This wouldn't be a const in real world code, but for display purposes here
+// to show the state object more clearly.
+const mockStateObject = {
+  subjects: {
+    items: ["art", "architecture"],
+  },
+  locations: {
+    items: ["sasb"],
+  },
+};
+
+// Will push ?subjects=art,architecture into browser url.
+router.push({
+  query: {
+    [multiSelectId]: selectedItems[multiSelectId].items.join(" "),
+  },
+});
+```
+
+This is an example with a handleChange function to manage the selectedItems state, without
+mutating it.
+
+```jsx
+import React, { useState } from "react";
+import MultiSelect from "./MultiSelect";
+
+function MultiSelectControlledExample() {
+  const [selectedItems, setSelectedItems] = useState({});
+
+  function handleChange(multiSelectId: string, itemId: string) {
+    let itemIds;
+    // Check if the id already exists in the state
+    if (selectedItems[multiSelectId] !== undefined) {
+      let itemIdExists =
+        selectedItems[multiSelectId].items.indexOf(itemId) > -1;
+      // Make a copy of the existing array.
+      itemIds = selectedItems[multiSelectId].items.slice();
+      // If itemId exists, remove it from the array.
+      if (itemIdExists) {
+        itemIds = itemIds.filter((id) => id != itemId);
+      } else {
+        // Add it to the array, but modify the copy, not the original.
+        itemIds.push(itemId);
+      }
+    } else {
+      itemIds = [];
+      itemIds.push(itemId);
+    }
+    setSelectedItems({
+      ...selectedItems,
+      [multiSelectId]: {
+        items: itemIds,
+      },
+    });
+  }
+
+  return (
+    <MultiSelect
+      id="subjects"
+      label="Subjects"
+      items={items}
+      selectedItems={selectedItems}
+      onChange={(selectedItem) => {
+        handleChange("id", selectedItem.id);
+      }}
+    />
+  );
+}
+
+export default MultiSelectControlledExample;
+```
+
+## useMultiSelect Hook
```

Comment:
There is a "Hooks" section in Storybook where this section belongs. Here is an example of the [useCarouselStyles](https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/story/hooks-usecarouselstyles--page) hook. You can still leave some documentation here and point to the specific `useMultiSelect` hook page.

Reply:
fixed